org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -129,6 +129,11 @@
 	 * Preference key to enable/disable downloading Maven source artifacts.
 	 */
 	public static final String MAVEN_DOWNLOAD_SOURCES = "java.maven.downloadSources";
+	/**
+	 * Preference key to enable/disable downloading source artifacts for Eclipse
+	 * projects.
+	 */
+	public static final String ECLIPSE_DOWNLOAD_SOURCES = "java.eclipse.downloadSources";
```

Comment:
Need a PR in the client side to expose the new preference key.

Reply:
I'm not sure if this is something we want to expose, simply because it looks like there's other cases where we don't, and I suspect it's because we want the feature to always be on. @fbricon , let me know your thoughts.

For `MavenBuildSupport`, and `InvisibleProjectBuildSupport`, discoverSource(..) seems to always execute if called.
For `EclipseBuildSupport`, discoverSource(..) only executes if "java.eclipse.downloadSources" is true.

So if Maven source downloading always works, then what does "java.maven.downloadSources" do ? Well, it looks like it's only used in MavenProjectImporter#importToWorkspace(..) to determine if all sources are downloaded on import, but it doesn't prevent on-demand downloading.

In other words, it seems like the intent is to always download sources for on-demand cases (eg. on hover, or go-to definition). If so, do we even need a "java.eclipse.downloadSources" ? It would only be useful if we needed to configure source downloading on initial import of an Eclipse project, but we don't support that yet.

Reply:
we need to keep it disabled by default (for eclipse projects) and expose a preference, because, when enabled, it'll change .classpath files by adding absolute paths to source files.
Invisble projects need to have the feature enabled regardless of this preference

Reply:
Makes sense, we should avoid messing with "user-controlled" .classpath if possibe. I'll expose the option in vscode-java and have it disabled by default.

Reply:
Exposed property in https://github.com/redhat-developer/vscode-java/pull/1758